lib/IRGen/GenCall.cpp

```diff
@@ -1696,7 +1700,8 @@ static void emitDirectExternalArgument(IRGenFunction &IGF, SILType argType,
 
   // Check to see if we can pairwise coerce Swift's exploded scalars
   // to Clang's expanded elements.
-  if (canCoerceToSchema(IGF.IGM, expandedTys, inputSchema)) {
+  if (AI.getCanBeFlattened() &&
```

Comment:
Do we need a check for AI.isDirect() here?

This can be called from a AI.isExtend() context.

Reply:
Oh, I missed that, I'll add that.